sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSender.java

```diff
@@ -651,13 +651,13 @@ private void completeClose() {
      * @param error Error to pass to pending sends.
      */
     private void handleError(Throwable error) {
-        final String logMessage = isDisposed.getAndSet(true)
-            ? "This was already disposed. Dropping error."
-            : "Disposing pending sends with error.";
-        logger.verbose("connectionId[{}] entityPath[{}] linkName[{}] {}", handler.getConnectionId(), entityPath,
-            getLinkName(), logMessage, error);
-
         synchronized (pendingSendLock) {
+            final String logMessage = isDisposed.getAndSet(true)
+                ? "This was already disposed. Dropping error."
+                : String.format("Disposing of %s pending sends with error.", pendingSendsMap.size());
```

Comment:
nit: `%d` instead of `%s`, or could this use String concatenation if this is being used on a hot path as `String.format` is much more expensive due to parsing requirements.